libs/core/langchain_core/language_models/structured_output.py

```diff
@@ -0,0 +1,18 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Generic, TypeVar, Union
+
+from langchain_core.language_models import LanguageModelInput
+from langchain_core.pydantic_v1 import BaseModel
+from langchain_core.runnables import Runnable
+
+_OutputSchema = TypeVar("_OutputSchema")
+
+
+class StructuredOutputMixin(Generic[_OutputSchema], ABC):
```

Comment:
we are back to NOT putting this on base class? why not?

Reply:
will only be documented on classes that implement it

Reply:
so the downside of putting on base class (and defaulting to not implemented) is that it will show up in the documentation for all classes? that doesnt seem that bad...

Reply:
what's downside of mixin?

Reply:
It's not that it shows up in documentation, but that it's hard to discover which models actually implement the interface.

After we implement, this we'll need to figure out how users will discover that this is an option